File: Java/exercicio12.java
class Produto {
    double preco;

    Produto(double preco) {
        this.preco = preco;
    }

    Produto somar(Produto outro) {
        return new Produto(this.preco + outro.preco);
    }
}

public class Main {
    public static void main(String[] args) {
        Produto produto1 = new Produto(50);
        Produto produto2 = new Produto(30);
        Produto produto3 = produto1.somar(produto2);
        System.out.println(produto3.preco);
    }
}

File: Go/exercicio12.go
package main

import "fmt"

type Produto struct {
    Preco float64
}

func SomarProdutos(p1, p2 Produto) Produto {
    return Produto{Preco: p1.Preco + p2.Preco}
}

func main() {
    produto1 := Produto{Preco: 50}
    produto2 := Produto{Preco: 30}
    produto3 := SomarProdutos(produto1, produto2)
    fmt.Println(produto3.Preco)
}
